nats.go

```diff
@@ -1807,6 +1808,24 @@ func (nc *Conn) makeTLSConn() error {
 	return nil
 }
 
+// TLSConnectionState retrieves the state of the TLS connection to the server
+func (nc *Conn) TLSConnectionState() (tls.ConnectionState, error) {
```

Comment:
Should this be passed by value here? Is that the recommended approach?

Reply:
Just looked seems so.

Reply:
yeah i find it awkward, i considered calling the go internal and returning a pointer which seemed a bit more idiomatic. So I am not sure if this is best, happy to do that pointer dance if you prefer

Reply:
Their API does same so we are good IMO.